### ISSUE TITLE
Requirements file not found 4 out of 5 upload code attempts

### ISSUE THREAD

**Issue by michaelschem** [aws/aws-toolkit-jetbrains#1962]
**Describe the bug**
About every 4 out of 5 times I try to "upload function code" I get the following error below.
![image](https://user-images.githubusercontent.com/2836129/89060552-10a2c500-d318-11ea-8612-dc24af2f216c.png)
I was thinking it was an issue with a resource using files in the project folder either by python or a terminal.  I closed both the terminal and any running instances of the project and it still seems to happen randomly, but frequently.  The only mitigation I've found is just trying it over and over again until it starts working.

**To reproduce**
1. Upload function code from AWS explorer to a lambda
2. 4 out of 5 times it will display the error above

**Expected behavior**
The code should upload with no error

**Screenshots**
see above

**Your Environment**
- OS: mac OS Catalina 10.15.4
- JetBrains product: PyCharm
- JetBrains product version: 2020.1.1
- AWS Toolkit version: 1.17-201
- SAM CLI version:1.0.0
- JVM/Python version:3.7

**Comment by abrooksv** [comment]
Does the generated template.yaml's codeUri look correct?

**Comment by michaelschem** [comment]
It does, is that actually used though?  Because my env variables on the lambda are not what I have in the template.

**Comment by michaelschem** [comment]
I'm thinking this actually may be related to the size of my project.  I previously had a bunch of large executables in my repo, but I've since moved them and I haven't had the issue yet.  I'll keep testing and add back in some large files later to see if the issue comes back.

**Comment by abrooksv** [comment]
Yes, it gets used. Env Vars defined in the IDE run config override what is defined in the template.

**Comment by michaelschem** [comment]
Oh, yea I figured that much, but it's not being used when I deploy to the lambda I'm assuming.  Which is where I have my problems.

**Comment by michaelschem** [comment]
It just threw the same error even without making my repo bigger, so I'm assuming that's not the problem :/

**Comment by abrooksv** [comment]
Can you make a minimal reproduction example?

**Comment by michaelschem** [comment]
oh, wow that could have been it. I wiped my .idea folder a bit ago b/c I was having another issue.  I've updated the project structure to this:
![image](https://user-images.githubusercontent.com/2836129/89076005-4a81c480-d334-11ea-8029-de64c590e440.png)
I'll keep uploading and see if the error comes up again.

**Comment by michaelschem** [comment]
...aaaand now it's creating the same error when I run locally, it seems to work when deploying

**Comment by michaelschem** [comment]
getting the same issue again on a new project.  For the first 15 times I built it, it worked just fine.  Then randomly (after a one line code change), it started throwing this error again.
![image](https://user-images.githubusercontent.com/2836129/90551671-88f5ec80-e146-11ea-9a97-c1957cda7d54.png)

**Comment by abrooksv** [comment]
Is the run configuration configured to use a template or handler directly?

**Comment by michaelschem** [comment]
In this case, directly. The other is using a template.

**Comment by abrooksv** [comment]
For the direct one, can you paste the CodeUri path from the template.yaml?

**Comment by michaelschem** [comment]
`CodeUri: .`, but I'm not using the template.yaml at all (thought that's what you mean by `handle directly`
![image](https://user-images.githubusercontent.com/2836129/90555018-8cd83d80-e14b-11ea-994f-4ee6dab4a365.png)

**Comment by abrooksv** [comment]
We generate a template on the fly when in "raw" mode. It seems the codeUri is wrong. It should point back into your main source, not the build dir.

**Comment by abrooksv** [comment]
I think it may be finding the handler in the wrong spot.

 Workaround would be to copy the dummy template as a sibling file to your requirements.txt and use that for now.

**Comment by michaelschem** [comment]
I think I already have that hierarchy

This is my current hierarchy.  Is that what you're talking about?

```
% tree -L 2
.
├── app.py
├── requirements.txt
├── template.yaml
├── test
│   ├── FunctionsTester.py
│   ├── SessionManager.py
│   ├── __init__.py
│   └── lambda.py
├── testing
│   ├── Test_SessionManager.py
│   └── __init__.py
└── venv
    ├── bin
    ├── include
    ├── lib
    └── pyvenv.cfg
```
